model/parVarSheet.cc

```diff
@@ -77,32 +77,21 @@ namespace minsky
   
 namespace
 {
-  std::string definition(const string& valueId)
+ 
+  std::string definition(const VariablePtr v)
   {
     SystemOfEquations system(cminsky());	  
-	ostringstream o;
-
-    for (const VariableDAG* i: system.variables)
-      {
-		if (i->valueId==valueId) 
-		{  		  
-           if (dynamic_cast<const IntegralInputVariableDAG*>(i) ||
-               !i || i->type==VariableType::constant) continue;
-           if (i->rhs)
-             o << i->rhs->matlab();
-	    }
-      }
-
-    for (const VariableDAG* i: system.integrationVariables)
-      {
-		if (i->valueId==valueId) 
-		{  
-            VariableDAGPtr input=system.expressionCache.getIntegralInput(i->valueId);
-            if (input && input->rhs)
-              input->rhs->matlab(o);
-	    }
-      }
-   
+    ostringstream o;
+	
+	auto varDAG=system.getNodeFromVar(*v);
+    
+    if (v->type()!=VariableType::parameter)
+    {
+      if (varDAG && varDAG->rhs && varDAG->type!=VariableType::constant && !dynamic_cast<const IntegralInputVariableDAG*>(varDAG.get()))
```

Comment:
It would suffice to check the type tag for being VariableType::integral, rather than dynamic_casting.